Feature

tGodson opened the pull request.

imhilla committed:
added README.md and sylelinter.config.js

imhilla committed:
Update stylelint.config.js

imhilla committed:
create stickler.yml

imhilla committed:
Update README.md

tGodson committed:
Added the HTML and CSS files

imhilla committed:
created the astronomy section

has the most popular section and editors pick

imhilla committed:
added footer section

imhilla committed:
added most popular links

imhilla committed:
added most popular section

imhilla committed:
added divs of the editor's pick and more in space

imhilla committed:
created the img folder

imhilla committed:
added divs for the more in space and astronomy

imhilla committed:
added images of the first row

imhilla committed:
added images for the second row

imhilla committed:
added images on the editor's section

tGodson committed:
Added the header

imhilla committed:
used grid to style the images

tGodson committed:
Added some Styled the header

- made the header look beautiful
- added icons, header logo and the left buttons on the header

imhilla committed:
styled the section

imhilla committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

imhilla committed:
fixed image issue

the logo took the image property pf height auto and 100% width changed it

imhilla committed:
added both footer horizontal lines

tGodson committed:
Added the first section of the body

- this section contains a heading, a photo and a little paragraph

tGodson committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

tGodson committed:
Modified the text in the p tag in the first section

imhilla committed:
displayed the bottom logo correctly

imhilla committed:
updated footer

imhilla committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

imhilla committed:
added the horizontal links

imhilla committed:
added the vertical line

tGodson committed:
Added the footer

imhilla committed:
resized the footer links

imhilla committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

imhilla committed:
fixed footer

imhilla committed:
matched font-family

imhilla committed:
fixed font-size

imhilla committed:
compressed gray color style

imhilla committed:
added the correct images

fixed font-size

imhilla committed:
fixed margin

fixed the hr line and made the closer
styled the hr and gave the margins

imhilla committed:
added a link styles

imhilla committed:
fixed horizontal to have links in it

imhilla committed:
fixed footer margin

imhilla committed:
enabled hoover effect on the links

gave it a gray color

imhilla committed:
fixed hover issues with the a tag

imhilla committed:
Update style.css

tGodson committed:
Added some part of the body

- used some CSS to style the body
- added some social media icons

tGodson committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

tGodson committed:
Finished added content to the page

tGodson committed:
Gave the footer a better structure

imhilla committed:
updated the readme

imhilla committed:
solved html validator error

imhilla committed:
Update README.md

imhilla committed:
Update style.css

imhilla committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

imhilla committed:
Update style.css

tGodson committed:
Made some correction after using the validator

imhilla committed:
Update style.css

tGodson committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

tGodson committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

imhilla committed:
Update style.css

imhilla committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

tGodson committed:
Added stickler and linter to my repo

tGodson committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

sumancrest0001 commented:
### STATUS: Require changes.
Awesome work TEAM :clap: :clap: 

**The page layout looks awesome.**

[Require] We have to fix one validation error in the HTML file.
![Screenshot from 2020-04-04 21-13-57](https://user-images.githubusercontent.com/37747619/78455256-ec620f00-76bc-11ea-93bd-2b5e0f2f609d.png)

sumancrest0001 commented:
[REQUIRED] You have a lot of repeated margin-left, margin-right in your CSS file. You can write one line code for that `margin: top right bottom left`. You can also define a class for `font-family: nyt-franklin, arial, helvetica, sans-serif;` and add that class where ever it is required. These steps to keep your code tidy.

sumancrest0001 commented:
[Required] :confused: :confused: Can you please tell us why you used **node modules**? If you want to use it, please add it to the .gitignore file and push the project to GitHub.

sumancrest0001 commented:
[REQUIRED] We have to also fix a couple of issues related to the layout of the page. Please make the following changes. After these changes, it will look similar to that of the original site.

Please add Editor's pick section beside the main section. [Requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/positioning_and_floating_elements/aside_requirement.png)

In the original site :point_down:  :point_down: 
![Screenshot from 2020-04-04 21-56-05](https://user-images.githubusercontent.com/37747619/78455620-2207f780-76bf-11ea-91fe-ef1b63029e1a.png)

sumancrest0001 commented:
The main logo should be a link not only an image. 
![Screenshot from 2020-04-04 21-30-34](https://user-images.githubusercontent.com/37747619/78455650-63000c00-76bf-11ea-8b36-42dab127f9d6.png)

sumancrest0001 commented:
Text of **The Theory of Inflation** section is not aligned as that of the original one.

**In your site.** :point_down:  :point_down: 
![Screenshot from 2020-04-04 21-37-46](https://user-images.githubusercontent.com/37747619/78455710-aa869800-76bf-11ea-951d-3a4abb158887.png)

Should be :point_down: 
![Screenshot from 2020-04-04 21-37-58](https://user-images.githubusercontent.com/37747619/78455719-b70af080-76bf-11ea-8e94-7cf49efca5bf.png)

sumancrest0001 commented:
Social media and other icons are not visible. Please check them out and fix them.
![Screenshot from 2020-04-04 21-30-49](https://user-images.githubusercontent.com/37747619/78455780-f9ccc880-76bf-11ea-9294-7dca5ccdd9d8.png)

sumancrest0001 commented:
Since this project focus on the alignment of items and positioning, we have to make sure that the positioning of each element is perfect. The Footer section looks a little bit different from the original one. Please make the following changes.

in your project :point_down: :point_down: 
![Screenshot from 2020-04-04 21-34-51](https://user-images.githubusercontent.com/37747619/78455874-62b44080-76c0-11ea-88ff-bce4f59a1f45.png)


Should be :point_down: 
![Screenshot from 2020-04-04 21-35-02](https://user-images.githubusercontent.com/37747619/78455885-765fa700-76c0-11ea-9b60-fd1b6699d15e.png)

sumancrest0001 commented:
Fix these issues. :point_down: :point_down: 
in your project :point_down:  :point_down: 

![Screenshot from 2020-04-04 21-35-15](https://user-images.githubusercontent.com/37747619/78455930-a8710900-76c0-11ea-87a1-06a245a21d94.png)
should be :point_down: 
![Screenshot from 2020-04-04 21-35-32](https://user-images.githubusercontent.com/37747619/78455935-b1fa7100-76c0-11ea-90ef-5585ea98ac6e.png)

[TIPS] Reduce space between elements.

sumancrest0001 commented:
Align the small texts from right not left.
**In your project.**
![Screenshot from 2020-04-04 21-33-35](https://user-images.githubusercontent.com/37747619/78455972-f259ef00-76c0-11ea-9fd7-b41c8afe2a9e.png)

Should be :point_down: 
![Screenshot from 2020-04-04 21-33-48](https://user-images.githubusercontent.com/37747619/78455983-fe45b100-76c0-11ea-95a6-f55d3495152d.png)

sumancrest0001 commented:
There is a footer section that should be sticky at the bottom all the time. It is missing in your project. 
**In the original project.**

![Screenshot from 2020-04-04 22-12-36](https://user-images.githubusercontent.com/37747619/78456079-6eeccd80-76c1-11ea-80e2-791e0b4958cd.png)

sumancrest0001 commented:
**Kindly make the above changes and submit them for another review. :pray: :pray: :pray:**

Need help? Reach out on slack handle **Suman Shrestha**

imhilla committed:
fixed image descriptions

they are now on the left side

imhilla committed:
fixed footer margin

They now match that of newyork times

imhilla committed:
fixed footer

used display of flex and not grid template columns

imhilla committed:
fixed sticky footer

imhilla committed:
updated the footer

tGodson committed:
made the changed required by TSE

tGodson committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

tGodson committed:
made some corrections after using the html validator

sumancrest0001 commented:
### STATUS: changes required

You have done nice improvements :+1: :+1: :clap: 
But there are still a couple of things that need to be fixed as suggested in the previous review.

**Please fix the following issues. :point_down:**

sumancrest0001 commented on index.html:
```diff
@@ -0,0 +1,671 @@
+            <img src="./img/search.png" alt="search" />
+            <a class="space-cosmos">Space & Cosmos</a>
+        </nav>
+        <nav>
+            <img src="./img/logo.PNG" alt="logo" />
+        </nav>
```
[REQUIRED] Please add <a> tag on the logo image. It should be link also not only an image.

sumancrest0001 commented:
[REQUIRED] Please .gitignore file and add node modules on that so that It is not pushed to remote repository. It makes your repo bulky.

sumancrest0001 commented:
[Required] You have added another footer section which is fine but we want you to add a sticky footer section as stated in the previous review. Please have a close look.

**In the original site :point_down:  :point_down:** 
![Screenshot from 2020-04-05 09-39-04](https://user-images.githubusercontent.com/37747619/78466495-5b714f00-7721-11ea-9da4-d1a0eae5221f.png)

the last section is a sticky footer. [reference](https://www.nytimes.com/2014/03/18/science/space/detection-of-waves-in-space-buttresses-landmark-theory-of-big-bang.html)

sumancrest0001 commented:
[Required] You have fixed text alignment issues with the footer section but there is an overlap between contents in the subscribe section.
Have a look :point_down: 
![Screenshot from 2020-04-05 09-41-50](https://user-images.githubusercontent.com/37747619/78466563-1699e800-7722-11ea-9ccb-1253c0238dd1.png)

sumancrest0001 commented:
[Required] Social media icons are still not visible. Please fix this issue also.
![Screenshot from 2020-04-05 09-45-25](https://user-images.githubusercontent.com/37747619/78466577-4ba63a80-7722-11ea-924f-bae5d1569550.png)

sumancrest0001 commented:
[Important change] In the HTML file you have used font-family as class`<p class="font-family">By LARRY BUCHANAN and JONATHAN CORUM</p>` but defined as **id** in the CSS file  `#font-family {
  font-family: nyt-franklin, arial, helvetica, sans-serif;
}`

sumancrest0001 commented:
**Kindly make the above changes and resubmit for another review. :+1: :+1: 
Happy coding.**

imhilla committed:
Update style.css

imhilla committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

imhilla committed:
added link on the logo

imhilla committed:
Update style.css

imhilla committed:
solved the sticky part

imhilla committed:
Update index.html

imhilla committed:
Create .gitignore

tGodson committed:
made the following changes

- made footer sticky
- made the logo to be a link
- added some css to make the web page look exactly like the reference
- did all that the TSE recommended

tGodson commented:
the social media icons are displaying on my screen

tGodson committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

christianotieno reviewed (commented):
## Pull Request under review by Christian Otieno (TSE)

christianotieno commented on README.md:
```diff
@@ -0,0 +1,55 @@
+# Positioning and Floating Elements
+This project consisted of making a clone of a New York Times article page.The New York Times has been a leader in experimenting with the inclusion of multimedia content in traditional stories. Which creates all kinds of interesting position questions.
+Here's the original page http://www.nytimes.com/2014/03/18/science/space/detection-of-waves-in-space-buttresses-landmark-theory-of-big-bang.html?_r=0
+
+![screenshot](./form-screenshot.png)
+
```
## [REQUIRED ]- FIX - The README
Your README should match the [standards](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/) that have been set in the pathwright. It is among the [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) for your repo to get approved.
##
- For some reason, most of your images are not being displayed properly. An example can be found in your README for instance.
![README](https://user-images.githubusercontent.com/22345961/78498826-5c23d880-7755-11ea-9604-be7b33af2f74.png)

I presume you meant this: 👇 ?
```suggestion
![screenshot](./app_screenshot.png)
```


This and among the many, my colleague has highlighted in the previous changes as well. What they are seeing is exactly what I seeing on my end. They are working perfectly well in localhost but not in production.

christianotieno replied:
We also don't have a live link to the site in your README 
- You can use [githack](https://raw.githack.com/) to set up a live link and embed it in your README.

christianotieno replied:
## [REQUIRED] - FIX - The positioning of the More in Space and Astronomy section
The 'More in Space and Astronomy' section should be fixed to match the original version;
- This is what we currently have: 👇 
![position_clone](https://user-images.githubusercontent.com/22345961/78499886-17e80680-775c-11ea-84aa-fdd065d9c669.png)
 - THis is what we should be aiming to achieve: 👇 
![position](https://user-images.githubusercontent.com/22345961/78499908-39e18900-775c-11ea-8e6f-60e7e1fbdf8a.png)
Resize it.

christianotieno replied:
## [REQUIRED] - FIX - The footer
We should also fix the footer as well.
- This is what we have currently: 👇
![footer_clone](https://user-images.githubusercontent.com/22345961/78499986-9d6bb680-775c-11ea-810e-ef8d83e6fe71.png)
- We should aim to have it look something close to this: 👇 
![footer](https://user-images.githubusercontent.com/22345961/78499996-b2e0e080-775c-11ea-8b0f-69ea20322a00.png)

christianotieno replied:
## [OPTIONAL] - REPLACE - The emended image and video.
I applaud the good job done into building this project.
- We can always try and improve on some areas that might make our site even more appealing.
- Consider these changes optional as the general minimum requirements have already been met in these sections. 
##
Our embedded video: 👇 
![stanford_clone](https://user-images.githubusercontent.com/22345961/78500086-30a4ec00-775d-11ea-8fcf-cb16d022af1e.png)
We may change it to match the original: 👇 
![stanford_professor](https://user-images.githubusercontent.com/22345961/78500095-3dc1db00-775d-11ea-91c1-b8e2218c1718.png)
Our embedded image: 👇 
![bicept_clone](https://user-images.githubusercontent.com/22345961/78500104-49150680-775d-11ea-9d8e-2d27d95758d9.png)

We may change it to match the original: 👇 
![bicept](https://user-images.githubusercontent.com/22345961/78500114-58944f80-775d-11ea-97e4-ec114e1b65d6.png)

christianotieno reviewed (changes requested):
# **CHANGES REQUIRED** :warning:
Hey @tGodson & @imhilla   :wave:
-
You have done a splendid job! :tada: Keep it up.
We, however, do have some few changes to carry out before we proceed (see review comment).
##
In addition to my colleague's previous changes, below are some small issues I found while reviewing your awesome project.

Please makes changes and submit another review request.
- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
- Again when you have made the specified changes, please submit another code review. Thank you!
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

tGodson committed:
made some changes recommended by the TSE

imhilla committed:
Update style.css

imhilla committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

imhilla committed:
Update style.css

imhilla committed:
Update style.css

imhilla committed:
Update style.css

tGodson committed:
change the extensions of the icons which were not dispaying

tGodson committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

tGodson committed:
added the life demo link

imhilla committed:
Update style.css

imhilla committed:
Merge branch 'feature' of https://github.com/tGodson/The-New-York-Times into feature

uryelah commented:
Hello **Hillary Odhiambo** & **Godson Tendongze Zeingkeang**, I'm Sarah, your last reviewer for this PR 👩‍🚀  

Great job working on all the fixes pointed out by the last TSE! 

[IMPORTANT] Sorry that this was not noted before but remember to add, and to keep, a descriptive Pull request message and title. Like the one in the gif below:
![69548761-3a8d3600-0f98-11ea-9f19-14da361c12e2](https://user-images.githubusercontent.com/16193212/78522994-b3ea3000-77a5-11ea-955b-3c7a0c717e9b.gif)

I'll mark this as optional since it was not noted in the previous review, but please remember to add the PR message in your next projects.

------
## Your project is approved! 🎊 🥅 💛 
### 
**You may close this Pull Request and merge the changes**

----

### Optional Changes

- [ ] Add a descriptive PR message.

- [ ] The navbar and fixed footer should take 100% of the screen width. Sometimes you can miss that according to your screen size. A fast trick to see how your page looks in a slightly wider screen is clicking `ctrl+`, or you could use your browser responsiveness tools.

----

**Keep being awesome devs and good luck with your next project!**

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)
